File: algorithms/563.cpp
#include <iostream>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
private:
    int helper(TreeNode *node, int &result) {
        if (node == NULL)
            return 0;
        int left = helper(node->left, result);
        int right = helper(node->right, result);
        result += abs(left-right);
        return left + right + node->val;
    }
public:
    int findTilt(TreeNode *root) {
        int result = 0;
        helper(root, result);
        return result;
    }
};

int main() {
    return 0;
}

File: algorithms/python3/563.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def findTilt(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.rls = 0
        self.helper(root)
        return self.rls

    def helper(self, root):
        if not root:
            return 0
        left = self.helper(root.left)
        right = self.helper(root.right)
        self.rls += abs(left - right)
        return left + right + root.val
